akka-stream/src/main/boilerplate/akka/stream/scaladsl/ZipLatestWithApply.scala.template

```diff
@@ -0,0 +1,109 @@
+/**
+ * Copyright (C) 2014-2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+package akka.stream.scaladsl
+
+import akka.stream._
+import akka.stream.impl.StreamLayout
+import akka.stream.stage.{ GraphStage, GraphStageLogic, InHandler, OutHandler }
+
+trait ZipLatestWithApply {
+
+  [2..20#/**
+   * Create a new `ZipLatestWith` specialized for 1 inputs.
+   *
+   * @param zipper zipping-function from the input values to the output value
+   */
+  def apply[[#A1#], O](zipper: ([#A1#]) ⇒ O): ZipLatestWith1[[#A1#], O] =
+    new ZipLatestWith1(zipper)
+  #
+
+  ]
+
+}
+
+[2..20#/** `ZipLatestWith` specialized for 1 inputs */
+class ZipLatestWith1[[#A1#], O] (val zipper: ([#A1#]) ⇒ O) extends GraphStage[FanInShape1[[#A1#], O]] {
+  override def initialAttributes = Attributes.name("ZipLatestWith1")
+  override val shape: FanInShape1[[#A1#], O] = new FanInShape1[[#A1#], O]("ZipLatestWith1")
+  def out: Outlet[O] = shape.out
+  [#val in0: Inlet[A1] = shape.in0#
+  ]
+
+  override def createLogic(inheritedAttributes: Attributes): GraphStageLogic = new GraphStageLogic(shape) {
+      // Without this field the completion signalling would take one extra pull
+      var willShutDown = false
+
+      [#var hasValue0 = false#
+      ]
+      [#var value0 : A1 = _#
+      ]
+      var hasLast  = false
+      [#var last0: A1 = _#
+      ]
+      var waitingForTuple = false
+
+      setHandler(
+        out,
+        new OutHandler {
+          override def onPull(): Unit = {
+            if (hasAllValues) {
+              pushIfFirstTupleOrDifferentTuple()
+            } else {
+              waitingForTuple = true
+            }
+            pullAllIfNeeded()
+          }
+        }
+      )
+
+      [#setHandler(
+        in0,
+        new InHandler {
+          override def onPush() = {
+            value0 = grab(in0)
+            hasValue0 = true
+            if (waitingForTuple && hasAllValues) {
+                pushTuple()
+                waitingForTuple = false
+                pullAllIfNeeded()
```

Comment:
so it turns out that 'waitingForTuple' indicates that a pull has been done downstream, and 'hasAllValues' simply indicates that we have all values necessary to form up a tuple. 'hasAllValues' stays 'true' after emission of the first tuple. Therefore, even without pulling upstream here, a pull downstream will trigger emission of the next tuple (since 'onPull' triggers 'pullAllIfNeeded')